examples/kwok/kwok_with_config/kwok_test.go

```diff
@@ -0,0 +1,135 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kwok
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/e2e-framework/pkg/envconf"
+	"sigs.k8s.io/e2e-framework/pkg/features"
+)
+
+func TestKwokCluster(t *testing.T) {
+	deploymentFeature := features.New("appsv1/deployment").
+		Setup(func(ctx context.Context, t *testing.T, cfg *envconf.Config) context.Context {
+			// setup node
+			node := newNode("kwok-node")
+			if err := cfg.Client().Resources().Create(ctx, node); err != nil {
+				t.Fatal(err)
+			}
+			// start a deployment
+			deployment := newDeployment(cfg.Namespace(), "test-deployment", 1)
+			if err := cfg.Client().Resources().Create(ctx, deployment); err != nil {
+				t.Fatal(err)
+			}
+			time.Sleep(2 * time.Second)
+			return ctx
+		}).
+		Assess("deployment creation", func(ctx context.Context, t *testing.T, cfg *envconf.Config) context.Context {
+			var dep appsv1.Deployment
+			if err := cfg.Client().Resources().Get(ctx, "test-deployment", cfg.Namespace(), &dep); err != nil {
+				t.Fatal(err)
+			}
+			if &dep != nil {
+				t.Logf("deployment found: %s %s", dep.Name, cfg.Namespace())
+			}
+
+			return context.WithValue(ctx, "test-deployment", &dep)
+		}).
+		Assess("pod ready", func(ctx context.Context, t *testing.T, cfg *envconf.Config) context.Context {
+			pods := &corev1.PodList{}
+			err := cfg.Client().Resources(cfg.Namespace()).List(context.TODO(), pods)
+			if err != nil || pods.Items == nil {
+				t.Error("error while getting pods", err)
+			}
+			if len(pods.Items) != 1 {
+				t.Fatal("could not find any pod in the namespace")
+			}
+			pod := pods.Items[0]
+			for _, condition := range pod.Status.Conditions {
+				if condition.Type == corev1.PodReady && condition.Status != corev1.ConditionTrue {
+					t.Fatal("pod is not ready")
+				}
+			}
+			return ctx
```

Comment:
This could be replaced with `PodConditionMatch` from the `pkg/wait/conditions` I believe right ?

Reply:
agreed. fixed it